src/libutil/executable-path.cc

```diff
@@ -0,0 +1,67 @@
+#include "environment-variables.hh"
+#include "executable-path.hh"
+#include "strings-inline.hh"
+#include "util.hh"
+#include "file-path-impl.hh"
+
+namespace nix {
+
+namespace fs = std::filesystem;
+
+constexpr static const OsStringView pATH_separator =
```

Comment:
the capitalization seems kinda weird here... :D

Reply:
I think we only uppercase macros and types so far.

Reply:
path_var_separator then, you already do `PATH_VAR_SEP` too

Reply:
Did that